services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs

```diff
@@ -360,23 +362,30 @@ impl SentinelS2L2aCogsMetaData {
         params: &T,
         page: u32,
     ) -> Result<StacCollection> {
-        let client = reqwest::Client::new();
-        let text = client
-            .get(&self.api_url)
-            .query(&params)
-            .query(&[("page", &page.to_string())])
-            .send()
-            .await
-            .context(error::Reqwest)?
-            .text()
-            .await
-            .context(error::Reqwest)?;
+        let client = Client::builder().build()?;
+
+        // TODO: configure max retries and delay, also find good defaults
+        retry(3, 125, 2., async || -> Result<StacCollection> {
```

Comment:
I think the magic numbers should be config params. Either for the provider or globally.

Reply:
a8c60bf